pennylane/gradients/vjp.py

```diff
@@ -318,7 +318,8 @@ def processing_fn(results):
             vjp_ = processing_fns[t_idx](res_t)
 
             if vjp_ is None:
-                vjps.append(None)
+                if reduction == "append":
+                    vjps.append(None)

```

Comment:
This solves the bug

>A bug was present when batch executing a list of tapes that may contain tapes with no parameters. To fix this, an if statement was added to the vjp function. Corresponding tests to check for this behaviour have also been added to all three interface tests.

Reply:
If the autodiff framework uses an `"extend"` reduction of the VJP vector, we want to skip appending `None` for tapes that have no parameter.